pkg/controller/bundlec_test/delay_start_delete_removed_object_test.go

```diff
@@ -0,0 +1,51 @@
+package bundlec_test
+
+import (
+	"context"
+	"testing"
+
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/smith/pkg/controller/bundlec"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Should initiate a delay for deletion controlled object that is not in the bundle
+func TestDelayDeleteStart(t *testing.T) {
+	t.Parallel()
+	m1 := configMapNeedsUpdate()
+	if m1.Annotations == nil {
+		m1.Annotations = make(map[string]string)
+	}
+	m1.Annotations["smith.atlassian.com/deletionDelay"] = "1h"
+	tc := testCase{
+		mainClientObjects: []runtime.Object{
+			m1,
+		},
+		bundle: &smith_v1.Bundle{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:       bundle1,
+				Namespace:  testNamespace,
+				UID:        bundle1uid,
+				Finalizers: []string{bundlec.FinalizerDeleteResources},
+			},
+		},
+		expectedActions: sets.NewString("PUT=/api/v1/namespaces/" + testNamespace + "/configmaps/" + mapNeedsAnUpdate),
+		appName:         testAppName,
+		namespace:       meta_v1.NamespaceAll,
+		testHandler: fakeActionHandler{
+			response: map[path]fakeResponse{
```

Comment:
I was also going to test that the request body contains a `deletionTimestamp` annotation, for which I need to intercept the request body, and after reading it I would have to re-create some implementation of the `io.ReadCloser`, so that the next handler in the chain will be able to read it.
I've found an `httputil.DumpRequest` that does that (https://rominirani.com/golang-tip-capturing-http-client-requests-incoming-and-outgoing-ef7fcdf87113), but it returns a raw formatted string instead of a structured result :( We can still the implementation code from this function if we want, but for now I decided not to bother...